Volumey/ViewModel/Settings/AppVolumeSyncViewModel.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using log4net;
+using Volumey.CoreAudioWrapper.Wrapper;
+using Volumey.DataProvider;
+using Volumey.Model;
+
+namespace Volumey.ViewModel.Settings
+{
+	public sealed class AppVolumeSyncViewModel : INotifyPropertyChanged
+	{
+		private int masterVolume = 100;
+		public int MasterVolume
+		{
+			get => masterVolume;
+		}
+        private bool isOn;
+		public bool IsOn
+		{
+			get => isOn;
+			set
+			{
+				isOn = value;
+				if(isOn)
+				{
+					this.deviceProvider.DefaultDeviceChanged += OnDefaultDeviceChanged;
+					this.defaultDevice = deviceProvider.DefaultDevice;
+					if(this.defaultDevice != null)
+					{
+                        this.masterVolume = this.defaultDevice.Master.Volume;
+                        this.defaultDevice.ProcessCreated += OnProcessCreated;
+						this.defaultDevice.Master.PropertyChanged += OnMasterPropertyChanged;
+					}
+				}
+				else
+				{
+					this.deviceProvider.DefaultDeviceChanged -= OnDefaultDeviceChanged;
+					if(this.defaultDevice != null)
+					{
+						this.defaultDevice.ProcessCreated -= OnProcessCreated;
+					}
+					this.defaultDevice = null;
+				}
+				if(SettingsProvider.Settings.SyncAppVolumeIsOn != value)
+				{
+					Task.Run(() =>
+					{
+						if(isOn)
+							Logger.Info($"Synchronising Application volume to Master volume on change.");
+						SettingsProvider.Settings.SyncAppVolumeIsOn = value;
+						_ = SettingsProvider.SaveSettings();
+					});
+				}
+				OnPropertyChanged();
+			}
+		}
+
+		private readonly IDeviceProvider deviceProvider;
+		private OutputDeviceModel defaultDevice;
+
+		private ILog _logger;
+		private ILog Logger => _logger ??= LogManager.GetLogger(typeof(AppVolumeSyncViewModel));
+
+		public AppVolumeSyncViewModel()
+		{
+			this.deviceProvider = DeviceProvider.GetInstance();
+			this.IsOn = SettingsProvider.Settings.SyncAppVolumeIsOn;
+		}
+
+        private void SyncSessionVolume()
+        {
+			if (this.IsOn)
+			{
+				foreach (var process in this.defaultDevice.Processes)
+				{
+					if (process.Volume != this.MasterVolume)
+					{
+						process.SetVolume(this.MasterVolume, false, ref GuidValue.Internal.VolumeGUID);
```

Comment:
I think better use the Volume property to change the volume to keep consistency, since you use this property in OnProcessCreated and SetVolume is mostly used for hotkeys specific scenarios.